shared/src/store/reduce-exercise-state.ts

```diff
@@ -32,6 +36,10 @@ export function applyAction(
     draftState: Mutable<ExerciseState>,
     action: ExerciseAction
 ) {
+    if (isDevelopment) {
+        // Make sure that the action isn't mutated in the reducer
+        freeze(action, true);
```

Comment:
Maybe this would lead to worse performance, but wouldn't it be way easier to do a `cloneDeepMutable` of the whole `action` here?